src/hooks/useTokenBalances.ts

```diff
@@ -85,25 +85,25 @@ async function _getBalances(walletInfo: WalletInfo, tokens: TokenDetails[]): Pro
   assert(contractAddress, 'No valid contract address found. Stopping.')
 
   const balancePromises: Promise<TokenBalanceDetails | null>[] = tokens.map((token) => {
-    const timeoutPromise = timeout<TokenBalanceDetails | null>({
+    const cacheKey = constructCacheKey({ token, userAddress, contractAddress, networkId })
+
+    // timoutPromise == Promise<never>, correctly determined to always throw
+    const timeoutPromise = timeout({
       timeoutErrorMsg: 'Timeout fetching balances for ' + token.address,
     })
 
     const fetchBalancesPromise = fetchBalancesForToken(token, userAddress, contractAddress, networkId).then(
       (balance) => {
-        const cacheKey = constructCacheKey({ token, userAddress, contractAddress, networkId })
         balanceCache[cacheKey] = balance
         return balance
       },
     )
 
+    // balancePromise == Promise<TokenBalanceDetails | never> == Promise<TokenBalanceDetails> or throws
```

Comment:
remove no?

Reply:
I added it specifically to explain stuff, as it's not very obvious how `never` type works

Reply:
gotcha, cool!